### PR TITLE
feat(notification): add idempotency in notification api

### PR DESCRIPTION
**Pull request by mabdh**
- add a new idempotency table to track idempotencies
- notify receiver API accepts optional idempotency-key header, if exists request is idempotent
- add a job to clean up idempotencies data (by default 24h)
- refactor some telemetries and cli

### CODE REVIEW

**Comment by krtkvrm** [comment]
can you please add details on PR?

File: core/notification/mocks/idempotency_repository.go [hunk 1]
```diff
@@ -0,0 +1,163 @@
+// Code generated by mockery v2.14.0. DO NOT EDIT.
+
+package mocks
```
Comment (krtkvrm) [inline core/notification/mocks/idempotency_repository.go:3]: why not have just _mock.go in same pkg ?

File: core/notification/handler.go [hunk 2]
```diff
@@ -86,10 +87,10 @@ func (h *Handler) Process(ctx context.Context, runAt time.Time) error {
 	if len(receiverTypes) == 0 {
 		return errors.New("no receiver type plugin registered, skipping dequeue")
 	} else {
-		ctx, span := h.messagingTracer.StartSpan(ctx, "batch_dequeue", trace.StringAttribute("messaging.handler_id", h.identifier))
+		traceCtx, span := h.messagingTracer.StartSpan(ctx, "batch_dequeue", trace.StringAttribute("messaging.handler_id", h.identifier))
```
Comment (krtkvrm) [inline core/notification/handler.go:90]: is tracectx is different than ctx ?

File: core/notification/mocks/idempotency_repository.go (hunk 1 above)
Comment (ravisuhag) [reply core/notification/mocks/idempotency_repository.go:3]: I think it is better to place it in mocks only to keep generated code out of the core package.

File: internal/api/header.go [hunk 3]
```diff
@@ -0,0 +1,25 @@
+package api
+
+import (
+	"context"
+
+	"google.golang.org/grpc/metadata"
+)
+
+const HeaderKeyIdempotency = "Idempotency-Key"
```
Comment (krtkvrm) [inline internal/api/header.go:9]: how about this can be configurable ?

File: core/notification/handler.go (hunk 2 above)
Comment (mabdh) [reply core/notification/handler.go:90]: yes it is different, renaming it just to surpress linting warning

File: internal/server/server.go [hunk 4]
```diff
@@ -33,9 +33,10 @@ import (
 const defaultGracePeriod = 5 * time.Second
 
 type Config struct {
-	Host          string `mapstructure:"host" yaml:"host" default:"localhost"`
-	Port          int    `mapstructure:"port" yaml:"port" default:"8080"`
-	EncryptionKey string `mapstructure:"encryption_key" yaml:"encryption_key" default:"_ENCRYPTIONKEY_OF_32_CHARACTERS_"`
+	Host          string            `mapstructure:"host" yaml:"host" default:"localhost"`
+	Port          int               `mapstructure:"port" yaml:"port" default:"8080"`
+	EncryptionKey string            `mapstructure:"encryption_key" yaml:"encryption_key" default:"_ENCRYPTIONKEY_OF_32_CHARACTERS_"`
+	APIHeaders    api.HeadersConfig `mapstructure:"api_headers" yaml:"api_headers""`
```
Comment (rahmatrhd) [inline internal/server/server.go]:
```suggestion
	APIHeaders    api.HeadersConfig `mapstructure:"api_headers" yaml:"api_headers"`
```

**Review by krtkvrm — APPROVED** [review]

**Pull request merged**